src/Sleep.js

```diff
@@ -0,0 +1,143 @@
+class Sleep {
+  constructor(userId, sleepData) {
+    this.userId = userId;
+    this.data = sleepData;
+    this.userData = this.extractData();
+  }
+
+  extractData() {
+    const userData = this.data.filter(user => user.userID === this.userId);
+    return userData;
+  }
+
+  findDay(date) {
+    const day = this.userData.find(dataPoint => dataPoint.date === date);
+    return day
+  }
+
+  calculateSum(arrayOfNumbers) {
+    return arrayOfNumbers.reduce((a, b) => a + b)
+  }
+
+  calculateAverage(arrayOfNumbers) {
+    const average = this.calculateSum(arrayOfNumbers) / arrayOfNumbers.length;
+    return Number(average.toFixed(1))
+  }
+
+  findDailyHours(date) {
+    const day = this.findDay(date)
+    if (day) {
+      return day.hoursSlept
+    } else {
+      return 'Sorry, you don\'t have any sleep information from that day'
+    }
+  }
+
+  calculateAverageHoursTotal() {
+    const sleepHours = this.userData.map(dataPoint => dataPoint.hoursSlept);
+    return this.calculateAverage(sleepHours)
+  }
+
+  calculateAverageHoursWeekly(endDate) {
+    const day = this.findDay(endDate);
+    if (day) {
+      const userData = this.userData.reverse()
+      const index = userData.indexOf(day);
+      const week = userData.splice(index, 7);
+      const weekHours = week.map(day => day.hoursSlept);
+      return this.calculateAverage(weekHours)
+    } else {
+      return 'Sorry, you don\'t have sleep data from all the days within the range you entered.'
+    }
+  }
+
+  findDailyQuality(date) {
+    const day = this.findDay(date)
+    if (day) {
+      return day.sleepQuality
+    } else {
+      return 'Sorry, you don\'t have any sleep information from that day'
+    }
+  }
+
+  calculateAverageQualityTotal() {
+    const sleepQualities = this.userData.map(dataPoint => dataPoint.sleepQuality);
+    return this.calculateAverage(sleepQualities)
+  }
+
+  calculateAverageQualityWeekly(endDate) {
```

Comment:
This is way better than splice (**yoink**). We are going to include the current day in our seven day spread now?